### DISCUSSION TITLE
Search for keyword in subject?

### DISCUSSION THREAD

**Discussion by datamanx** [mjs/imapclient#435] [Q&A] [answered]
Hello, 

Below returns unseen emails and their respective subject lines.  How do I only return unseen email subject lines that match a keyword?

Many thanks in advance....

     import email
    from imapclient import IMAPClient

    HOST = 'somthin.com'
    USERNAME = 'email'
    PASSWORD = 'pwd'

    with IMAPClient(HOST) as server:
    server.login(USERNAME, PASSWORD)
    server.select_folder("INBOX", readonly=True)
    
    messages = server.search("UNSEEN")
    for uid, message_data in server.fetch(messages, "RFC822").items():
        email_message = email.message_from_bytes(message_data[b"RFC822"])
        #print(uid, email_message.get("From"), email_message.get("Subject"))
        print(uid, email_message.get("Subject"))

**Comment by shoaib30** [comment, marked as answer]
you will have to chain the search criteria like

`connection.search(['UNSEEN', 'SUBJECT', 'test subject'])`
